docs/docs/user-guide/training-data-format.mdx

```diff
@@ -0,0 +1,659 @@
+  - <another synonym variation>
+```
+
+Synonyms can also be definedin-line in your training examples by
+specifying the `value` of the entity:
+
+```yaml
+nlu:
+- intent: check_balance
+  examples: |
+  - how much do I have on my [credit card account]{"entity": "account", "value": "credit"}
+  - how much do I owe on my [credit account]{"entity": "account", "value": "credit"}
+```
+
+To use the synonyms defined in your training data, you need to make sure the
+pipeline contains the [`EntitySynonymMapper`](../nlu/components#entitysynonymmapper) component. You
+should define synonyms when there are multiple ways users refer to the same
+thing.
+
+#### Example 
+
+For example, let's say you had an entity `account`, and you expect the
+value "credit". Your users also refer to their "credit" account as "credit
+account" and "credit card account".
+
+In this case, you could define "credit card account" and "credit account" as
+**synonyms** to "credit":
+
+```yaml
+- synonym: credit
+  examples: |
+  - credit card account
+  - credit account
+```
+
+Then, if either of these phrases is extracted as an entity, it will be
+mapped to the **value** `credit`.
+
+:::note
+Synonym mapping only happens **after** entities have been extracted.
+That means that you need to provide examples of the variations 
+on a synonym in your training examples so
+that Rasa Open Source can learn to pick them up.
+:::
+
+### Regular Expression Features
+
+Regular expressions can be used in two different ways:
+
+1. They can be used to support intent classification and entity
+extraction when using the [RegexFeaturizer](../nlu/components#regexfeaturizer) component in the pipeline.
+
+2. They can be used to extract entities from a user message using the rule-based
+[`RegexEntityExtractor`](../nlu/components#regexentityextractor) component in the pipeline.
+
+For example, if your entity has a deterministic structure (like a
+zipcode or an email address), you can use a regular expression to ease
+detection of that entity (using the `RegexFeaturizer`) or to
+extract the entities from the user message directly (using the
+`RegexEntityExtractor`). For a bank account number that is 10-12 digits long,
+it might look like this:
+
+```yaml
+- regex: account_number
+  examples: |
+  - \d{10:12}
+```
+
+If you are using regular expressions to directly extract entities using
+the `RegexEntityExtractor`, the name of the regular expression should
+match the name of the entity you want to extract.
+
+If you are using the regular expressions for the `RegexFeaturizer` the
+name of the regular expression does not matter. If does not define the
+entity nor the intent, it is a human readable description for you
+to remember what this regex is used for and is the title of the
+corresponding pattern feature.
+
+If you want to use the RegexFeaturizer you can also use the regex
+features to improve the intent classification performance, for example,
+by defining a greet clause:
```

Comment:
Do you think it's worth it to split this into 2 parts: regex for entity extraction and regex for intent classification?

Reply:
Then we can easily add info about L308-314, as it only applies to entity extraction and not to intent classification (as all intent classifiers use the features)

Reply:
the features also apply to entities though. How about 
```
#### Regular Expression Features

#### Regular Expressions for Rule-Based Entity Extraction
```
Or is that too long a subtitle?

Reply:
Oh oh wait I see what you mean. Yes your version makes more sense.